src/util/status_file_test.go

```
@@ -0,0 +1,186 @@
+	Context("when file becomes unreadable after creation", func() {
+		It("should return error", func() {
+			statusFilePath := filepath.Join(tempDir, "status.json")
+
+			// Create a valid file first
+			content := StatusFileContent{Status: StatusInitializing}
+			data, _ := json.Marshal(content)
+			err := os.WriteFile(statusFilePath, data, 0644)
+			Expect(err).ToNot(HaveOccurred())
+
+			// Make it unreadable
+			os.Chmod(statusFilePath, 0000)
```

golangci-lint (2.12.2)

[error] Error return value of `os.Chmod` is not checked

(errcheck)